contracts/FragToken.sol

```diff
@@ -64,20 +73,40 @@ contract FRAGToken is ERC20, ERC20Permit, Ownable {
 
         // add to current locked amount
         _locksAmount[msg.sender] = _locksAmount[msg.sender] + amount;
-        _locksBlock[msg.sender] = block.number;
+        _locksBlock[msg.sender] = block.timestamp;
+
+        if(period == Period.TwoWeeks)
+            _lockTime[msg.sender] = block.timestamp + (2 * _TIMELOCK);
+        
+        if(period == Period.OneMonth)
+            _lockTime[msg.sender] = block.timestamp + (4 * _TIMELOCK);
+
+        if(period == Period.ThreeMonths)
+            _lockTime[msg.sender] = block.timestamp + (12 * _TIMELOCK);
+
+        if(period == Period.SixMonths)
+            _lockTime[msg.sender] = block.timestamp + (24 * _TIMELOCK);
+
+        if(period == Period.OneYear)
+            _lockTime[msg.sender] = block.timestamp + (52 * _TIMELOCK);
```

Comment:
I think we need an if/elseif construct with a final else as I'm not 100% sure if solidity will handle unknown cases.
Transactions can be crafted with any bytes layout...

Reply:
Yes, correct.